includes/class-edd-db-discounts.php

```diff
@@ -0,0 +1,542 @@
+				$month = date( 'm', strtotime( $args['end_date'] ) );
+				$day   = date( 'd', strtotime( $args['end_date'] ) );
+
+				$where .= " AND $year = YEAR ( end_date ) AND $month = MONTH ( end_date ) AND $day = DAY ( end_date )";
+			}
+
+		}
+
+		// Specific paid date or in a paid date range
+		if ( ! empty( $args['start_date'] ) ) {
+
+			if ( is_array( $args['start_date'] ) ) {
+
+				if ( ! empty( $args['start_date']['start_date'] ) ) {
+
+					$start_date = date( 'Y-m-d H:i:s', strtotime( $args['start_date']['start_date'] ) );
+
+					$where .= " AND `start_date` >= '{$start_date}'";
+
+				}
+
+				if ( ! empty( $args['start_date']['end'] ) ) {
+
+					$end = date( 'Y-m-d H:i:s', strtotime( $args['start_date']['end'] ) );
+
+					$where .= " AND `start_date` <= '{$end}'";
+
+				}
+
+			} else {
+
+				$year  = date( 'Y', strtotime( $args['start_date'] ) );
+				$month = date( 'm', strtotime( $args['start_date'] ) );
+				$day   = date( 'd', strtotime( $args['start_date'] ) );
+
+				$where .= " AND $year = YEAR ( start_date ) AND $month = MONTH ( start_date ) AND $day = DAY ( start_date )";
+			}
+
+		}
+
+		// Specific search query
+		if ( ! empty( $args['search'] ) ) {
+			$where .= " AND ( name LIKE '%%" . $args['search'] . "%%' OR code LIKE '%%" . $args['search'] . "%%' )";
+		}
+
+		if ( ! empty( $where ) ) {
+			$where = ' WHERE 1=1 ' . $where;
+		}
+
+		return $where;
+	}
+
+	/**
+	 * Count the total number of discounts in the database.
+	 *
+	 * @access public
+	 * @since 3.0
+	 *
+	 * @param array $args {
+	 *      Arguments for the `WHERE` clause.
+	 *
+	 *      @type mixed array|string $type         Discount types (flat/percent).
+	 *      @type mixed array|string $status       Discount status (active/inactive/expired).
+	 *      @type mixed array|string $date_created Date created. Can pass a `start` and `end` key in the array to allow
+	 *                                             for further filtering.
+	 *      @type mixed array|string $end_date     Discount expiration date. Can pass a `start` and `end` key in the
+	 *                                             array to allow for further filtering.
+	 *      @type mixed array|string $start_date   Discount start date. Can pass a `start` and `end` key in the array to
+	 *                                             allow for further filtering.
+	 *      @type string             $search       Search parameters.
+	 * }
+	 *
+	 * @return int $count Number of discounts in the database.
+	 */
+	public function count( $args = array() ) {
+		global $wpdb;
+
+		$where = $this->parse_where( $args );
+		$sql   = "SELECT COUNT($this->primary_key) FROM " . $this->table_name . "{$where};";
+		$count = $wpdb->get_var( $sql );
+
+		return absint( $count );
+	}
+
+	/**
+	 * Count the number of discounts in the table and group by status.
+	 *
+	 * @access public
+	 * @since 3.0
+	 *
+	 * @return object Counts by status.
+	 */
+	public function count_by_status() {
```

Comment:
I think this method should be named `counts_by_status`. (plural) `count_by_status` implies a single count, where one would expect to pass a status to count. This is doing multiple counts.